Szorny.cpp

```diff
@@ -1,19 +1,50 @@
-#include "Szorny.h"
-
-
-int Szorny::getDmg()const {
-	return dmg;
-}
-
-int Szorny::getHp()const {
-	return hp;
-}
-
-std::string Szorny::getName()const {
-	return nev;
-}
-
-void Szorny::tamad(Szorny & a)const {
-	a.hp = a.hp - this->dmg;
-	if (a.hp < 0) a.hp = 0;
-}
+#include "Szorny.h"
+
+
+int Szorny::getDmg()const {
+	return dmg;
+}
+
+int Szorny::getHp()const {
+	return hp;
+}
+
+std::string Szorny::getName()const {
+	return nev;
+}
+
+void Szorny::tamad(Szorny & a)const {
+	a.hp = a.hp - this->dmg;
+	if (a.hp < 0) a.hp = 0;
+}
+
+Szorny Szorny::parseUnit(std::string fajlnev) {
+	int hp, dmg;
+	std::string nev;
+	std::vector<std::string> v;
+	std::ifstream f(fajlnev);
+	std::string sor = "";
+	while (!f.eof()) {
+		getline(f, sor);
+		v.push_back(sor);
+	}
+	f.close();
+	for (int i = 12; i < v[1].size() - 2; i++) {
+		sor = sor + v[1][i];
+	}

```

Comment:
Nagyon csúnyák ezek a varázskonstansok (12, -2, 9, 10, stb.). Igaz ugyan, hogy nem kértünk tökéletes input ellenőrzést, de azt tényleg nehéz elképzelni, hogy valaki a json fájlok megszerkesztésénél még a soron belüli oszloppozíciók megőrzésére is kínosan ügyelni fog. Sokkal szerencsésebb lenne a vezérlőjeleket keresni a karakterláncokban (idézőjelek, vesszők, kettőspont, ilyesmire gondolok), és az azonosított környezetben lévő adatokat kinyerni. Megtehetitek ezt később is, de előbb-utóbb úgyis foglalkozni kell majd ezzel.

A másik, hogy mennyivel jobb az érdekes részeket egy string-ekből álló vektorból kinyerni, mint abból a stringből, amibe a getline eleve olvasott?